libs/agent-sm/bootstrap/src/main/java/org/opensearch/javaagent/bootstrap/AgentPolicy.java

```diff
@@ -46,11 +100,15 @@ public static void setPolicy(Policy policy) {
      * @param trustedHosts trusted hosts
      * @param classesThatCanExit classed that are allowed to call {@link System#exit}
      */
-    public static void setPolicy(Policy policy, final Set<String> trustedHosts, final String[] classesThatCanExit) {
+    public static void setPolicy(
+        Policy policy,
+        final Set<String> trustedHosts,
+        final BiFunction<Class<?>, List<Class<?>>, Boolean> classesThatCanExit
```

Comment:
Using a function here so we could test it (otherwise the `AnyCanExit` (that is default) always passes